windows/src/engine/keyman32/tests/keyman-engine-tests/appinttests.cpp

```diff
@@ -6,7 +6,7 @@ TEST(AppContext, Get_Max_buff) {
   WCHAR callbuf[MAXCONTEXT];
   AppContext testContext;
   WCHAR *testString = L"ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789abcdefghijklmnopqrstuvwxyz0123456789";
-  WCHAR *expectedString = L"ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789abcdefghijklmnopqrstuvwxyz0"; // MAXCONTEXT taking into account null character termination
+  WCHAR *expectedString = L"JKLMNOPQRSTUVWXYZ0123456789abcdefghijklmnopqrstuvwxyz0123456789"; // MAXCONTEXT taking into account null character termination
```

Comment:
Why is the expected string truncating the beginning of the buffer? I thought it would truncate the end...

Reply:
The AppContext Class has an array of 16bit Unicode characters (CurContext) that stores the current context it is limited to 64. The Character at index 0 into the array is the furthest from the caret.  So when truncating we are interested in the text closet to the caret, therefore we truncate the begging of the buffer.